.golangci-kal.yml

```diff
@@ -16,6 +16,7 @@ linters:
               - "nofloats" # Ensure floats are not used.
               - "nomaps" # Ensure maps are not used.
               - "nophase" # Phase fields are discouraged by the Kube API conventions, use conditions instead.
+              - "optionalorrequired" # Every field should be marked as `+optional` or `+required`.
```

Comment:
Is that true?  I always assumed the default was `required`.  Although I did see how you were able to create a GRPCRoute without a `spec`.

Reply:
Yeah :/ the linter is more about making the intention explicit. 

I don’t remember the full historic context on the field requirement markers, but on past having a field with omitempty would cause it to be optional, or a pointer as well (IIRC, may be wrong) so the default wasn’t to be required, but how markers and json tags were added. 

Enforcing us to make the required/optional explicit avoids falling on the situation like above

Reply:
Here: https://github.com/zecke/Kubernetes/blob/master/docs/devel/api-conventions.md#optional-vs-required

“Note that for backward compatibility, any field that has the omitempty struct tag, and is not explicitly marked as +required, will be considered to be optional. This is expected to change in the future, and new fields should explicitly set either an +optional or +required comment tag.“

Reply:
Long live intention over implication. 🚀